src/main/java/org/kohsuke/github/Requester.java

```diff
@@ -23,368 +23,24 @@
  */
 package org.kohsuke.github;
 
-import com.fasterxml.jackson.databind.JsonMappingException;
-import org.apache.commons.io.IOUtils;
-import org.apache.commons.lang3.StringUtils;
-
-import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.io.InterruptedIOException;
-import java.io.Reader;
-import java.io.UnsupportedEncodingException;
-import java.lang.reflect.Array;
-import java.lang.reflect.Field;
-import java.net.HttpURLConnection;
-import java.net.MalformedURLException;
-import java.net.ProtocolException;
-import java.net.SocketException;
-import java.net.SocketTimeoutException;
-import java.net.URI;
-import java.net.URISyntaxException;
-import java.net.URL;
-import java.net.URLEncoder;
-import java.nio.charset.StandardCharsets;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.HashMap;
 import java.util.Iterator;
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.Locale;
-import java.util.Map;
-import java.util.NoSuchElementException;
-import java.util.Objects;
 import java.util.function.Consumer;
-import java.util.logging.Logger;
-import java.util.zip.GZIPInputStream;
 
-import javax.annotation.CheckForNull;
 import javax.annotation.Nonnull;
-import javax.annotation.WillClose;
-import javax.net.ssl.SSLHandshakeException;
-
-import static java.util.Arrays.asList;
-import static java.util.logging.Level.*;
-import static org.apache.commons.lang3.StringUtils.defaultString;
-import static org.kohsuke.github.GitHub.MAPPER;
-import static org.kohsuke.github.GitHub.connect;
 
 /**
- * A builder pattern for making HTTP call and parsing its output.
+ * A thin helper for {@link GitHubRequest.Builder} that includes {@link GitHubClient}.
```

Comment:
This is the really the summary of this change: Refactoring from `Requester` being monolithic center of work to having the work split between `GitHubClient`, `GitHubRequest`, and `GitHubResponse` among others.